drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -602,93 +654,107 @@
-        Validator receives all the public keys). The details of the mechanisms
-        responsible for key distribution are outside the scope of this
-        document.</t>
-
-        <t>This method MUST use an algorithm pair defined in <xref
-        target="IOAM-IP-registry"/> and the approach MUST be asymmetric.</t>
+        <t>The symmetric key based signature algorithm MUST use SHA-256
+        (<xref target="SHS"/>) as the Digest Algorithm, and MUST use Advanced
+        Encryption Standard (<xref target="AES"/>) with a key length of 256 bits
+        and the Galois/Counter Mode (<xref target="NIST.800-38D"/>) as the
+        Signature Algorithm (AES-256-GCM). The corresponding Signature Suite
+        Identifier is 1, as defined in <xref target="IOAM-IP-registry"/>. As a
+        consequence, the signature consumes 32 octets and is computed according
+        to the following steps:
+
+          <list style="numbers">
+            <t>The encapsulating node creates a nonce and stores it in the Nonce
+            field of the IOAM Integrity Protection header (the Nonce length
+            field is set accordingly). The Signature-suite field is set to 1.
+            The signature is generated over the hash of header fields (see
+            <xref target="NewIOAMOptionTypes"/> for the exact list of header
+            fields to include in the signature, depending on the IOAM
+            Option-Type) and IOAM-Data-Fields it has inserted, i.e.,
+            sign(hash(Header-Fields || IOAM-Data-Fields)), with the Nonce field
+            provided as the nonce. IOAM-Data-Fields supposed to be modified by
+            other IOAM nodes on the path MUST be excluded from the signature
+            (e.g., the POT Cumulative field). The signature is stored in the
+            Signature field of the IOAM Integrity Protection header. Important
+            note: if all the inserted IOAM-Data-Fields are supposed to be
+            modified by other IOAM nodes on the path, or if there is no
+            IOAM-Data-Field inserted at all, then the encapsulating node MUST
+            NOT use an Integrity Protected Option-Type.</t>
+
+            <t>A transit node generates a signature over the hash of
+            IOAM-Data-Fields it has inserted, i.e., sign(hash(IOAM-Data-Fields)),
+            with the Signature field provided as the nonce. IOAM-Data-Fields
+            modified in-place by the transit node MUST be excluded from the
+            signature (e.g., the POT Cumulative field). The signature is stored
+            in the Signature field of the IOAM Integrity Protection header.
+            Important note: if the transit node does not insert IOAM-Data-Fields
+            (e.g., it only modifies IOAM-Data-Fields in-place, or does nothing),
+            then the transit node MUST NOT generate a signature and MUST NOT
+            update the Signature field.</t>
+
+            <t>The decapsulating node (i.e., the Validator) is responsible for
```

Comment:
Looking at how the text reads, we might want to expand on the "validator" a bit, e.g.,

The decapsulating node performs the validation of the integrity protection. It is referred to as "Validator" in the context of the integrity protection method. The Validator is responsible for ...